find is not working

I've successfully installed the ejdb ruby bindings according to the installation guide.
Creating a new database/collection works, but when I try to execute the "One snippet intro" the following exception is raised when it comes to find the data:

```
results = jb.find("parrots", {"likes" => "toys"}, {"$orderby" => {"name" => 1}})
TypeError: wrong argument type Hash (expected Data)
```

The same exception is raised with ```find_one```

Here's my setup:

- Mac OS X 10.8.5
- Ruby 2.0 p247
- Installed ejdb via homebrew (v1.1.24)

I have a very similar setup:
* Mac OS X 10.10.2
* ruby 2.0.0p598 (2014-11-13 revision 48408) [x86_64-darwin14.1.0] (installed via ``rvm``)
* ejdb: stable 1.2.5 (installed via homebrew)

``make check`` returns the following:
```
$ make check
make -C ./test
ruby t1.rb
t1.rb:37:in `find': wrong argument type Hash (expected Data) (TypeError)
	from t1.rb:37:in `<main>'
make[1]: *** [check] Error 1
make: *** [check] Error 2
```

--- a/ext/rbejdb/src/rbejdb.c
+++ b/ext/rbejdb/src/rbejdb.c
@@ -502,8 +502,8 @@ VALUE EJDB_find_internal(VALUE self, VALUE collName, VALUE queryWrap, VALUE q, V
 }
 
 VALUE EJDB_find_internal_wrapper(VALUE args) {
-    return EJDB_find_internal(rb_ary_pop(args), rb_ary_pop(args), rb_ary_pop(args),
-                              rb_ary_pop(args), rb_ary_pop(args), rb_ary_pop(args));
+    return EJDB_find_internal(rb_ary_entry(args, 0), rb_ary_entry(args, 1), rb_ary_entry(args, 2),
+                              rb_ary_entry(args, 3), rb_ary_entry(args, 4), rb_ary_entry(args, 5));
 }
 
 VALUE EJDB_find_ensure(VALUE queryWrap, VALUE exception) {

--- a/rbejdb.gemspec
+++ b/rbejdb.gemspec
@@ -17,4 +17,5 @@ Gem::Specification.new do |spec|
     --exclude .*\.o
     --exclude .*\.so
   ]
+  spec.add_development_dependency 'test-unit', '~> 3.0', '>= 3.1.2'
 end

